src/main/kotlin/org/rust/lang/core/resolve/LangItems.kt

```diff
@@ -35,13 +36,39 @@ fun findIteratorItemType(project: Project, ty: RustType): RustType {
     return rawType.substitute(typeParameterMap)
 }
 
+fun findIndexOutputType(project: Project, containerType: RustType, indexExprType: RustType): RustType {
+    val impls = RsImplIndex.findImpls(project, containerType)
+    for (impl in impls) {
+        val trait = impl.traitRef?.resolveToTrait ?: continue
+        if (trait.isIndex) {
+            // TODO: get index type from impl declaration
+            val indexType = impl.functionList
```

Comment:
Ideally, we should be able to get type from `Type Output = ` if we fix the logic over there: https://github.com/intellij-rust/intellij-rust/blob/8ae5457c32c48e10d8b99ee17aeb112ac204abf6/src/main/kotlin/org/rust/lang/core/types/infer/Declarations.kt#L17